packages/manager/apps/pci-object-storage/src/domain/regions/regionFilter.spec.ts

```diff
@@ -0,0 +1,572 @@
+import { describe, expect, it } from 'vitest';
+import { TRegionAvailability } from '@ovh-ux/manager-pci-common';
+import {
+  filterAllowedRegions,
+  groupRegionsByContinent,
+  ProductAvailability,
+  ProjectRegion,
+  RegionFilterCriteria,
+} from './regionFilter';
+import { OBJECT_CONTAINER_OFFER_STORAGE_STANDARD } from '@/constants';
+
+describe('filterAllowedRegions', () => {
```

Comment:
General comments for all the file :
There should not be comment explaining what you are testing inside your tests.
There should not be complicated calculation for your expect. It looks like you are rewriting the implementation of each method when you do things like `result.forEach(region => truc.filter(...))`. Unit tests for fonction should be a much as possible a simple `expect(result).toEqual(expected)`and I think it's possible in most cases here.